website/content/en/docs/getting-started-with-terraform/_index.md

```diff
@@ -0,0 +1,392 @@
+instance profile.
+
+```hcl
+data "aws_iam_policy" "ssm_managed_instance" {
+  arn = "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"
+}
+
+resource "aws_iam_role_policy_attachment" "karpenter_ssm_policy" {
+  role       = module.eks.worker_iam_role_name
+  policy_arn = data.aws_iam_policy.ssm_managed_instance.arn
+}
+
+resource "aws_iam_instance_profile" "karpenter" {
+  name = "KarpenterNodeInstanceProfile-${var.cluster_name}"
+  role = module.eks.worker_iam_role_name
+}
+```
+
+Go ahead and apply the changes.
+
+```bash
+terraform apply -var cluster_name=$CLUSTER_NAME
+```
+
+Now, Karpenter can use this instance profile to launch new EC2 instances and 
+those instances will be able to connect to your cluster.
+
+### Create the KarpenterController IAM Role
+
+Karpenter requires permissions like launching instances, which means it needs
+an IAM role that grants it access. The config below will create an AWS IAM 
+Role, attach a policy, and authorize the Service Account to assume the role 
+using [IRSA](https://docs.aws.amazon.com/emr/latest/EMR-on-EKS-DevelopmentGuide/setting-up-enable-IAM.html).
+We will create the ServiceAccount and connect it to this role during the Helm
+chart install.
+
+```hcl
+module "iam_assumable_role_karpenter" {
+  source                        = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"
+  version                       = "4.7.0"
+  create_role                   = true
+  role_name                     = "karpenter-controller-${var.cluster_name}"
+  provider_url                  = module.eks.cluster_oidc_issuer_url
+  oidc_fully_qualified_subjects = ["system:serviceaccount:karpenter:karpenter"]
+}
+
+resource "aws_iam_role_policy" "karpenter_contoller" {
+  name = "karpenter-policy-${var.cluster_name}"
+  role = module.iam_assumable_role_karpenter.iam_role_name
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = [
+          "ec2:CreateLaunchTemplate",
+          "ec2:CreateFleet",
+          "ec2:RunInstances",
+          "ec2:CreateTags",
+          "iam:PassRole",
+          "ec2:TerminateInstances",
+          "ec2:DescribeLaunchTemplates",
+          "ec2:DescribeInstances",
+          "ec2:DescribeSecurityGroups",
+          "ec2:DescribeSubnets",
+          "ec2:DescribeInstanceTypes",
+          "ec2:DescribeInstanceTypeOfferings",
+          "ec2:DescribeAvailabilityZones",
+          "ssm:GetParameter"
+        ]
+        Effect   = "Allow"
+        Resource = "*"
+      },
+    ]
+  })
+}
+```
+
+Since we've added a new module, you'll need to run `terraform init` again. 
+Then, apply the changes.
+
+```bash
+terraform init
+terraform apply -var cluster_name=$CLUSTER_NAME
+```
+
+### Create the EC2 Spot Service Linked Role
+
+This step is only necessary if this is the first time you're using EC2 Spot in this account. More details are available [here](https://docs.aws.amazon.com/batch/latest/userguide/spot_fleet_IAM_role.html).
```

Comment:
Can we remove this now that the default is on demand again?

Reply:
I had it in there because the CF-based guide had it. But, went ahead and removed it.

Reply:
I mean to tag @ellistarn with this question but ty

Reply:
SGTM